torchgeo/datasets/south_africa_crop_type.py

```diff
@@ -0,0 +1,278 @@
+      "Crop Type Classification Dataset for Western Cape, South Africa",
+      Version 1.0, Radiant MLHub, https://doi.org/10.34911/rdnt.j0co8q
+
+    .. versionadded:: 0.6
+    """
+
+    filename_regex = r"""
+        ^(?P<field_id>[0-9]*)
+        _(?P<date>[0-9]{4}_[0-9]{2}_[0-9]{2})
+        _(?P<band>(B[0-9A-Z]{2} | VH | VV))
+        _10m"""
+    date_format = "%Y_%m_%d"
+    rgb_bands = ["B04", "B03", "B02"]
+    s1_bands = ["VH", "VV"]
+    s2_bands = [
+        "B01",
+        "B02",
+        "B03",
+        "B04",
+        "B05",
+        "B06",
+        "B07",
+        "B08",
+        "B8A",
+        "B09",
+        "B11",
+        "B12",
+    ]
+    all_bands: list[str] = s1_bands + s2_bands
+    cmap = {
+        0: (0, 0, 0, 255),
+        1: (255, 211, 0, 255),
+        2: (255, 37, 37, 255),
+        3: (0, 168, 226, 255),
+        4: (255, 158, 9, 255),
+        5: (37, 111, 0, 255),
+        6: (255, 255, 0, 255),
+        7: (222, 166, 9, 255),
+        8: (111, 166, 0, 255),
+        9: (0, 175, 73, 255),
+    }
+
+    def __init__(
+        self,
+        paths: Union[str, Iterable[str]] = "data",
+        crs: Optional[CRS] = None,
+        classes: list[int] = list(cmap.keys()),
+        bands: list[str] = all_bands,
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+    ) -> None:
+        """Initialize a new South Africa Crop Type dataset instance.
+
+        Args:
+            paths: paths directory where dataset can be found
+            crs: coordinate reference system to be used
+            classes: crop type classes to be included
+            bands: the subset of bands to load
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+
+        Raises:
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+        """
+        assert (
+            set(classes) <= self.cmap.keys()
+        ), f"Only the following classes are valid: {list(self.cmap.keys())}."
+        assert 0 in classes, "Classes must include the background class: 0"
+
+        self.paths = paths
+        self.classes = classes
+        self.ordinal_map = torch.zeros(max(self.cmap.keys()) + 1, dtype=self.dtype)
+        self.ordinal_cmap = torch.zeros((len(self.classes), 4), dtype=torch.uint8)
+
+        super().__init__(paths=paths, crs=crs, bands=bands, transforms=transforms)
+
+        # Map chosen classes to ordinal numbers, all others mapped to background class
+        for v, k in enumerate(self.classes):
+            self.ordinal_map[k] = v
+            self.ordinal_cmap[v] = torch.tensor(self.cmap[k])
+
+    def __getitem__(self, query: BoundingBox) -> dict[str, Any]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and labels at that index
+        """
+        assert isinstance(self.paths, str)
```

Comment:
It's probably possible to relax this constraint by using `filepaths` to compute the correct directory for the mask. But let's do that another day, I want to get this merged.